src/Utilities.ts

```diff
@@ -327,32 +308,66 @@ export class Utilities {
      * Handles cleaning up the Sarif Viewer temp directory used for temp files (embedded code, converted files, etc.)
      */
     public static removeSarifViewerTempDirectory(): void {
-        const path = Utilities.Path.join(Utilities.Os.tmpdir(), Utilities.SarifViewerTempDir);
-        Utilities.removeDirectoryContents(path);
-        Utilities.Fs.rmdirSync(path);
+        const tempPath = path.join(Utilities.Os.tmpdir(), Utilities.SarifViewerTempDir);
+        Utilities.removeDirectoryContents(tempPath);
+        fs.rmdirSync(tempPath);
+    }
+
+    /**
+     * Fixes up file schemed based paths to contain the proper casing for VSCode's URIs which are case sensitive.
+     * @param uri The URI for which to fix the casing.
+     */
+    public static fixUriCasing(uri: Uri): Uri {
+        // We can only support file-system scheme files.
+        if (uri.scheme.toUpperCase() !== "FILE") {
+            return uri;
+        }
+
+        let pathPartIndex: number = 0;
+        const pathParts: string[] = uri.fsPath.split(path.sep);
+        if (pathParts.length < 1) {
+            return uri;
+        }
+
+        // We assume that the "root" drive is lower-cased which is "usually" the case.
+        const fixedPathParts: string[] = [ pathParts[0].toLowerCase() + path.sep];
+
+        while (pathPartIndex + 1 < pathParts.length) {
+            const directoryToRead: string = path.join(...fixedPathParts);
```

Comment:
>path.join [](start = 44, length = 9)

You could replace all joins by building up a 'fixedPath' variable by concat'ing new parts as you encounter them (I assume each iteration of this single append would be more efficient than the join). I don't think this would compromise readability too much. I doubt the micro-optimization has measurable impact but if you agree it doesn't compromise readability, something to consider. Our pathological case here on windows is roughly MAX_PATH/2 (c:\a\b\c\etc.), so somewhere around ~130 parts. in C#, we could use the directory info object to avoid building out all the intermediate paths, not sufficiently knowledgeable here to know whether you have a TS/JS alternate...